Type/Password.pm

```diff
@@ -176,4 +172,12 @@ sub _to_hash_type_instance {
     return;
 }
 
+sub _to_hash_type_instance_or_die {
+    my($hashed) = @_;
+    my($hti) = _to_hash_type_instance($hashed);
+    b_die('invalid password hash')
```

Comment:
You might want to print the value in the b_die, and I would simplify:

`return _to_hash_type_instance($hashed) or b_die('invalid password hash=', $hashed)`

I think in this particular case, the code is clearer. I don't always like the `or` approach but I like removing temporaries more.

Reply:
Right, though you'd need to use `||`, not `or`.